tsdb/head_read.go

```diff
@@ -632,15 +679,23 @@ type boundedIterator struct {
 // If there are samples within bounds it will advance one by one amongst them.
 // If there are no samples within bounds it will return false.
 func (b boundedIterator) Next() chunkenc.ValueType {
-	for b.Iterator.Next() == chunkenc.ValFloat {
-		t, _ := b.Iterator.At()
+	for typ := b.Iterator.Next(); typ != chunkenc.ValNone; typ = b.Iterator.Next() {
+		var t int64
+		switch typ {
+		case chunkenc.ValFloat:
+			t, _ = b.Iterator.At()
+		case chunkenc.ValHistogram:
+			t, _ = b.Iterator.AtHistogram()
+		case chunkenc.ValFloatHistogram:
+			t, _ = b.Iterator.AtFloatHistogram()
+		}
```

Comment:
There's an `AtT()` method that could be used instead for all three cases since we only need a timestamp

https://github.com/prometheus/prometheus/blob/f711d71aa8318dbb39522ac26f8ee77cc373f77b/tsdb/chunkenc/chunk.go#L143